src/pages/PlanPage/hooks/usePlanPageData.ts

```diff
@@ -0,0 +1,62 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+type URLParams = {
+  owner: string
+  provider: string
+}
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      username: z.string().nullable(),
+      isCurrentUserPartOfOrg: z.boolean(),
+      numberOfUploads: z.number().nullable(),
+    })
+    .nullable(),
+})
+
+const query = `
+  query PlanPageData($username: String!) {
+    owner(username: $username) {
+      username
+      isCurrentUserPartOfOrg
+      numberOfUploads
+    }
+  }
+`
+
+export function usePlanPageData() {
+  const { owner, provider } = useParams<URLParams>()
+
+  const variables = {
+    username: owner,
+  }
+
+  return useQuery({
+    queryKey: ['PlanPageData', variables, provider, query],
```

Comment:
You should be able to remove `query` from here, cache keys should only be variables that either declared in the hook, or passed via args.